src/pods/movies/reducer/filter.ts

```diff
@@ -1,3 +1,33 @@
-import { createEmptyMovieFilter } from "../viewModel";
+import { actionsEnums } from "../../../common-app/actions";
+import {MovieFilter} from './model';
 
-export const filterReducer = (state = createEmptyMovieFilter(), action) => { return state};
\ No newline at end of file
+
+type FilterState = MovieFilter;
+
+const createDefaultState = (): FilterState => ({
+    title: '',
+    genere: '',
+    year: null,
+});
+
+export const filterReducer = (state = createDefaultState(), action) => { 
+    
+    switch (action.type) {
+        case actionsEnums.FILTER_REQUEST_COMPLETED:
+            return handleUpdateFilters(state, action.payload);
+    }
+    
+    return state;
+}
+
+const handleUpdateFilters = (state: FilterState, payload: MovieFilter): FilterState => {
+    if (state.genere !== payload.genere){
+        return {...state, genere: payload.genere};
+    }else if(state.title !== payload.title){
+        return {...state, title: payload.title};
+    }else if(state.year !== payload.year){
+        return {...state, year: payload.year};
+    }
+    
+    return state;
```

Comment:
This version should

In this version we will just overwrite the filter.

In the updatefield version (other case opened):

```typescript
const handleUpdateFilter = (state: Filterstate, fieldId : string, value : any) : FilterState => ({
  state,
  [fieldId]: value,
})
```